base/linalg/dense.jl

```diff
@@ -909,53 +932,65 @@ compute the inverse tangent. Otherwise, the inverse tangent is determined by usi
 
 # Examples
 ```jldoctest
-julia> atan(tan(ones(2, 2)))
-2×2 Array{Float64,2}:
-  0.291927  -0.708073
- -0.708073   0.291927
+julia> atan(tan([0.5 0.1; -0.2 0.3]))
+2×2 Array{Complex{Float64},2}:
+  0.5+1.38778e-17im  0.1+0.0im
+ -0.2+8.32667e-17im  0.3-5.55112e-17im
 ```
 """
-function atan(A::StridedMatrix{T}) where T
+function atan(A::StridedMatrix)
     if ishermitian(A)
         return full(atan(Hermitian(A)))
     end
-    return log((one(A) - im * A) \ (one(A) + im * A)) / 2im
+    SchurF = schurfact(complex(A))
+    U = im * UpperTriangular(SchurF.T)
+    R = full(log((I - U) \ (I + U)) / 2im)
+    return SchurF.Z * R * SchurF.Z'
 end
```

Comment:
In principle, we could save a few matrix allocations here.   But the code would get a lot uglier and I doubt it's worth the trouble — uses of matrix inverse trig functions are rare, and I doubt most users will care about saving a bit of memory here.  They can always be further optimized later on if the need arises.